codehost/github/projects.go

```diff
@@ -166,3 +204,77 @@ func (c *GithubClient) GetProjectFieldsByProjectNumber(ctx context.Context, owne
 
 	return fields, nil
 }
+
+func (c *GithubClient) GetPullRequestProjectV2ItemIDByNumber(ctx context.Context, owner, repo, projectID string, number int) (string, error) {
```

Comment:
Can we please add a comment before explaining what this function does?

```go
// GetPullRequestProjectV2ItemIDByNumber returns the project item ID for a pull request.
// Each item on a project board is represented by a project item.
```

Also, I'm not sure if the `byNumber` here helps. I would suggest to rename it:
```suggestion
func (c *GithubClient) GetPullRequestProjectV2ItemID(ctx context.Context, owner, repo, projectID string, prNum int) (string, error) {
```

WDYT?